languageFactories/language-factory-core/src/main/scala/cromwell/languages/util/ImportResolver.scala

```diff
@@ -227,21 +245,37 @@ object ImportResolver {
         }).contextualizeErrors(s"download $toLookup")
       }
 
-    private def getUri(toLookup: WorkflowSource): Either[NonEmptyList[WorkflowSource], ResolvedImportBundle] = {
-      implicit val sttpBackend = HttpResolver.sttpBackend()
-      val responseIO: IO[Response[WorkflowSource]] = sttp.get(uri"$toLookup").headers(headers).send()
-
-      // temporary situation to get functionality working before
+    private def getUri(toLookup: WorkflowUrl): Checked[ResolvedImportBundle] = {
+      // Temporary situation to get functionality working before
       // starting in on async-ifying the entire WdlNamespace flow
-      val result: Checked[WorkflowSource] = Await.result(responseIO.unsafeToFuture(), 15.seconds).body.leftMap { e =>
-        NonEmptyList(e.toString.trim, List.empty)
+      // Note: this will cause the calling thread to block for up to 30 seconds
+      // (15 for the auth header lookup, 15 for the http request)
+      val unauthedAttempt = getUriInner(toLookup, Map.empty)
+      val result = if (StatusCodes.NotFound == unauthedAttempt.code) {
+        val authHeaders = Await.result(fetchAuthHeaders(uri"$toLookup"), 15.seconds)
```

Comment:
I would honestly drop this to 5 seconds, if it's taking 15 then Github's down and there's no point tying up threads trying.

Reply:
In the very near future, behind the scenes this will call ECM to get the Github access token. With that in mind would it be better to keep this at 15s or would you still suggest lowering it to 5s?

Reply:
I think the issue is blocking rather than the particular services involved, so yeah, still mild preference for lowering.